buildSrc/src/main/java/org/elasticsearch/gradle/FileContentsTask.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.gradle;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+
+import org.gradle.api.DefaultTask;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.OutputFile;
+import org.gradle.api.tasks.TaskAction;
+
+/**
+ * Creates a file and sets it contents to something.
+ */
+public class FileContentsTask extends DefaultTask {
+
+    /**
+     * The file to be built.
+     */
+    @OutputFile
+    private File file;
+
+    @Input
+    private Object contents;
+
+
+    public void contents(Object contents) {
```

Comment:
We prefer adding the `set` prefix.
```suggestion
    public void setContents(Object contents) {
```

Also in this particular case I think we should also make a semantic change and type `contents` with `String`. This does mean that groovy will resolve GStrings sooner, but looking at the uses of this task that is ok and I would prefer the stronger typing, it's one of the reasons we convert in the first place.
We can always _lather_ add a version that takes `Supplier<String>` if we need it to be more dynamic.